libraries/Microsoft.Bot.Expressions/Memory/SimpleObjectMemory.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Expressions.Memory
+{
+    public class SimpleObjectMemory : IMemory
+    {
+        private object memory = null;
+        private int version = 0;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SimpleObjectMemory"/> class.
+        /// This wraps a simple object as IMemory.
+        /// </summary>
+        /// <param name="memory">the object to wrap.</param>
+        public SimpleObjectMemory(object memory)
+        {
+            this.memory = memory;
+        }
+
+        public static IMemory Wrap(object obj)
+        {
+            if (obj is IMemory)
+            {
+                return (IMemory)obj;
+            }
+
+            return new SimpleObjectMemory(obj);
+        }
+
+        public (object value, string error) GetValue(string path)
+        {
+            var parts = path.Split(".[]".ToCharArray(), StringSplitOptions.RemoveEmptyEntries);
```

Comment:
"a.1" would be parsed as the same with "a[1]". does it make sense?

Reply:
here the issue is, can we really rely "." vs "[]" to decide it's an index or a property. The answer is no, because  a[b] is also a valid expression.

Reply:
I see, we probably can do smarter, by combining the information of "." and "[]" with the object is list or not.